### PR TITLE
[NumberType] Fixed parsing issue in core Number.h.

### PR DESCRIPTION
**Pull request by aswaunni**
Fixed conversion issue by adding extra condition for Hexadecimal (hexadecimal base type number in 0x format was not getting parsed)

### CODE REVIEW

File: Source/core/Number.h [hunk 1]
```diff
@@ -452,7 +452,7 @@ namespace Core {
                 if ((Value == 0) && (*Text == '0') && (Base == BASE_UNKNOWN)) {
                     // Base change, move over to an OCTAL conversion
                     Base = BASE_OCTAL;
-                } else if ((Value == 0) && (toupper(*Text) == 'X') && (Base == BASE_OCTAL)) {
+                } else if ((Value == 0) && (toupper(*Text) == 'X') && ((Base == BASE_OCTAL) || (Base == BASE_HEXADECIMAL))) {
```
Comment (pwielders) [inline Source/core/Number.h]: I think this should be without the base == BASE_OCTAL. it should only check for (Base == BASE_HEXADECIMAL)

File: Source/core/Number.h [hunk 2]
```diff
@@ -559,7 +559,7 @@ namespace Core {
...
```
Comment (pwielders) [inline Source/core/Number.h]: See previous comment

File: Source/core/Number.h [hunk 3]
```diff
@@ -632,7 +632,7 @@ namespace Core {
...
```
Comment (pwielders) [inline Source/core/Number.h]: See previous comment.

File: Source/core/Number.h (hunk 1 above)
Comment (aswaunni) [reply Source/core/Number.h]: When the number is of the form 0x, first we are checking for '0' and Base is set to BASE_OCTAL and hence that check.

Comment (pwielders) [reply Source/core/Number.h]: You are right. Again a good fix.

**Pull request merged**